cargo-shuttle/src/args.rs

```diff
@@ -131,14 +131,18 @@ pub enum Command {
     Login(LoginArgs),
     /// Log out of the Shuttle platform
     Logout(LogoutArgs),
-    /// Generate shell completions
+    /// Generate shell completions and manpage
     Generate {
         /// Which shell
         #[arg(short, long, env, default_value_t = Shell::Bash)]
         shell: Shell,
```

Comment:
I think we need to make this variable optional. Otherwise:

```sh
$ cargo shuttle generate --manpage

error: invalid value '/bin/bash' for '--shell <SHELL>'
  [possible values: bash, elvish, fish, powershell, zsh]

For more information, try '--help'.
```

Reply:
> I think we need to make this variable optional. Otherwise:

You mean the `shell` option?

Reply:
When running `generate`, the `--shell <shell>` OR `--manpage` flag has to be given, but not both at the same time. There are flags to set in the clap macros for making this behavior (I wonder if the env loading on the shell arg messes this up?). The description could say "Generate shell completions or manpage".

It's almost as if they could be turned into subcommands rather than flags :thinking:, but that would be a breaking change.

Reply:
From my testing, the `--shell` option *must* be given, whether or not `--manpage` is there or not. Just specifying `--manpage` seems to error out like @orhun mentioned. :confused:

Reply:
Yeah, I would say let's make both `--shell` and `--manpage` optional and have a condition to generate them accordingly.

Reply:
As far as I understand, we may need to remove the `env` attribute from the `shell` option. Because otherwise I think it's picked up automatically by clap through the `SHELL` environment variable. And since we're making it `Option<Shell>`, we may also need to remove the `default_value_t` attribute.

Reply:
> As far as I understand, we may need to remove the env attribute from the shell option.

Oooh, that's why I got "error: invalid value '/bin/bash'" (I have `SHELL` set to `bash`) Good catch!

Alternatively we can rename it to `SHELL_ARG` or something.

> And since we're making it Option<Shell>, we may also need to remove the default_value_t attribute.

I think this is fine.

Reply:
I think removing the env parse on SHELL and default value is fine. Explicit arg makes it more fool proof.